ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthServer.java

```diff
@@ -269,8 +271,9 @@ static MessageData constructGetPooledTransactionsResponse(
       if (maybeTx.isEmpty()) {
         continue;
       }
+
       final BytesValueRLPOutput txRlp = new BytesValueRLPOutput();
-      maybeTx.get().writeTo(txRlp);
+      TransactionEncoder.encodeRLP(maybeTx.get(), txRlp, EncodingContext.POOLED_TRANSACTION);
```

Comment:
this line fixes the `Versioned hashes from blob transactions do not match expected values` error